src/api/app/controllers/webui/staging/workflows_controller.rb

```diff
@@ -1,7 +1,8 @@
 class Webui::Staging::WorkflowsController < Webui::WebuiController
   VALID_STATES_WITH_REQUESTS = [:acceptable, :accepting, :review, :testing, :building, :failed, :unacceptable].freeze
 
-  before_action :require_login, except: [:show]
+  # TODO: Remove this when we'll refactor kerberos_auth
+  before_action :kerberos_auth, except: [:show]
   before_action :set_project, only: [:new, :create]
   before_action :set_workflow_project, except: [:new, :create]
   before_action :set_staging_workflow, except: [:new, :create]
```

Comment:
The line just below this one should be changed to verify that `authorize` was called in `new`. I think you had this change before. It's somehow gone.

Reply:
I've added it back because [when we redirect](https://github.com/openSUSE/open-build-service/blob/bab35af63a6d1418424d37d3292662315a324068/src/api/app/controllers/webui/staging/workflows_controller.rb#L13) we do not call `authorize` hence the `verify_authorized` will error.
I suppose calling `authorize` is a duty of the redirected routed at that point, but if you prefer we can add back the `:new` and call `authorize` also on the redirected resource: `authorize @project.staging`.
Please note that we also have an automated test now so the `verify_authorize` is mostly an extra.

Apologies: I should've mentioned this explicitly to save 1 round trip, next time I'll be more explicit.

Reply:
Oh, I didn't think about that. It's not obvious at all at a first glance. I think it's better to be explicit here, even if it means calling `authorize` before the `redirect_to` call, so it would be `if authorize @project.staging` as you suggest.

No worries for not mentioning this, it's always a pleasure to review your PRs.

Reply:
Having the `authorize` inside the `if` is completely fine.